src/hooks/useFavouriteMessages.ts

```diff
@@ -0,0 +1,44 @@
+
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { useState } from "react";
+
+const favouriteMessageIds = JSON.parse(
+    localStorage?.getItem("io_element_favouriteMessages")?? "[]") as string[];
+
+export default function useFavouriteMessages() {
+    const [, setX] = useState<string[]>();
```

Comment:
I think you should be able to avoid this using `useLocalStorage()`

Reply:
you mean instead of `localStorage.getItem` i should use the `useLocalStorage`  hook?

Reply:
Also @SimonBrandner  please, i don't know why the i18n is failing, i tried to check if it's from any of my file but apparently not

Reply:
> Also @SimonBrandner please, i don't know why the i18n is failing, i tried to check if it's from any of my file but apparently not

Have you run `yarn i18n`

Reply:
No i didn't run that, let me try it

Reply:
![useLocal](https://user-images.githubusercontent.com/38439166/178976598-382ce695-e5aa-43cb-9297-1d8f648c8e03.JPG)

Reply:
apparently, the hook itself has never been used before when i tried to search it's use case in the project, i guess due to this reason. Also, i am only using local storage temporarily, the code gonna change later on when i move to accountdata

Reply:
This looks odd and probably shouldn't happen... @t3chguy, do you have any idea what is happening here?

Reply:
Ok, so it turns out `useLocalStorage` can't be used here as the localStorage value can be altered from multiple places - let me think about a better solution